storage/src/robocorp/storage/_requests.py

```diff
@@ -58,70 +58,71 @@ def extensive_log(level, msg, *args, **kwargs):
     return before_sleep_log(LOGGER, logging.DEBUG, exc_info=True)
 
 
-class RequestsHTTPError(HTTPError):
-    """Custom `requests` HTTP error with status code and message."""
+# Exported for ease-of-use
+Response = requests.Response
 
-    def __init__(
-        self,
-        *args,
-        status_code: int = 0,
-        status_message: str = "Error",
-        **kwargs,
-    ):
-        super().__init__(*args, **kwargs)
 
+class HTTPError(_HTTPError):
+    """Custom `requests` HTTP error with status code and reason"""
+
+    def __init__(self, message: str, status_code: int, reason: str):
+        super().__init__(message)
+        self.message = message
         self.status_code = status_code
-        self.status_message = status_message
+        self.reason = reason
+
+    def __str__(self):
+        return f"{self.status_code} {self.reason} - {self.message}"
 
 
 class Requests:
     """Wrapper over `requests` 3rd-party with error handling and retrying support."""
 
-    def __init__(self, route_prefix: str, default_headers: Optional[dict] = None):
+    def __init__(self, route_prefix: str = "", default_headers: Optional[dict] = None):
         self._route_prefix = route_prefix
         self._default_headers = default_headers
 
     def handle_error(self, response: requests.Response):
-        resp_status_code = response.status_code
-        log_func = LOGGER.critical if resp_status_code // 100 == 5 else LOGGER.debug
-        log_func("API response: %s %r", resp_status_code, response.reason)
+        http_status = f"{response.status_code} {response.reason!r}"
+        if 500 <= response.status_code < 600:
+            LOGGER.critical("Server error: %s", http_status)
+        elif 400 <= response.status_code < 500:
+            LOGGER.info("Client error: %s", http_status)
+        else:
+            LOGGER.debug("Response: %s", http_status)
 
         if response.ok:
             return
 
-        fields = {}
+        # Fallback values
+        status_code = response.status_code
+        reason = response.reason
+        message = response.text
+
         try:
             fields = response.json()
+
+            # For some reason we might still get a string from the deserialized
+            # JSON payload, possible due to some double encoding bug in CR
```

Comment:
Have no idea if this was fixed already, but was happening loooong time ago.

Reply:
I know, I remember it being a "feature" of some old API. It very much _should_ be fixed in CR instead of having to deal with it in the library.

Reply:
Yes, can be kept for safety precautions, as it does no harm.